k8s/upgrade/Cargo.toml

```diff
@@ -25,5 +26,6 @@ http = "0.2.8"
 async-trait = "0.1.64"
 serde = "1.0.152"
 serde_json = "1.0.93"
+thiserror = "1.0.38"
```

Comment:
@niladrih is using snafu, let's either use snafu or thiserror, not both please

Reply:
The upgrade crate and the upgrade-job crates are largely unrelated. No shared resources whatsoever. @tiagolobocastro do you mean this in the sense that using the same error crate reduces repo dependencies?

Reply:
They're not the same crate?? They should be in the same create IMHO, they're all about upgrade. A crate can have multiple targets.

Reply:
That is true, but development for each of them was started independently at different points in time. The k8s/upgrade-job crate is meant to replace the operators/upgrade crate, and the k8s/upgrade crate was extracted from the k8s/plugin crate after it had started getting bulky.

We can maybe put in a TODO or a GitHub issue with the integration plan.